### DISCUSSION TITLE
Is it possible to add an emulation layer for sbt plugins?

### DISCUSSION THREAD

**Discussion by He-Pin** [com-lihaoyi/mill#4047] [General]
How about adding an emulation layer that wraps an SBT plugin and lets it work as a mill plugin, eg the sbt-header plugin

**Comment by lihaoyi** [comment]
Probably not possible

**Comment by lefou** [comment]
It's a pity, that so many useful tools are developed as concrete build tool plugins only.

IMHO, The proper way for any build tooling is to build the tooling with a CLI or API first and as a bonus wrap it in some lightweight plugin for a concrete build tool. Developing some functionality inside a build-tool plugin, without some proper CLI to do the same thing without the build tool is rather bad practice and likely a sign for a non-diverse build tooling landscape.
